### PR TITLE
VTCCA-2901 Type of evidence selection is defaulting to 'Lease' option…

### PR DESCRIPTION
**Pull request by reddy-yattapu**
… on 'Upload other evidence' screen

### CODE REVIEW

File: app/controllers/propertyLinking/FileUploadController.scala [hunk 1]
```diff
@@ -37,22 +37,26 @@ import uk.gov.hmrc.play.HeaderCarrierConverter
...
                              val businessRatesAttachmentService: BusinessRatesAttachmentService
                            )(implicit val messagesApi: MessagesApi, val config: ApplicationConfig)
   extends PropertyLinkingController with BaseController with Utf8MimeTypes {
 
-  lazy val form = Form(single("evidenceType" -> EnumMapping(EvidenceType)))
+  lazy val form: Form[EvidenceType] = Form(single("evidenceType" -> EnumMapping(EvidenceType)))
+
+
+
+  //lazy val form: Form[EvidenceType] = Form(mapping("evidenceType" -> EnumMapping(EvidenceType)))(EvidenceType.fromName(_))(evidenceType => evidenceType.)
```
Comment (DanielConnelly) [inline app/controllers/propertyLinking/FileUploadController.scala]: commented code.

File: app/controllers/propertyLinking/FileUploadController.scala [hunk 2]
```diff
@@ -37,22 +37,26 @@ import uk.gov.hmrc.play.HeaderCarrierConverter
...
-
 abstract class FileUploadController (
                              val authenticated: AuthenticatedAction,
                              val withLinkingSession: WithLinkingSession,
                              val businessRatesAttachmentService: BusinessRatesAttachmentService
                            )(implicit val messagesApi: MessagesApi, val config: ApplicationConfig)
   extends PropertyLinkingController with BaseController with Utf8MimeTypes {
 
-  lazy val form = Form(single("evidenceType" -> EnumMapping(EvidenceType)))
+  lazy val form: Form[EvidenceType] = Form(single("evidenceType" -> EnumMapping(EvidenceType)))
```
Comment (DanielConnelly) [inline app/controllers/propertyLinking/FileUploadController.scala]: should this form not sit inside EvidenceType companion object?

File: app/controllers/propertyLinking/UploadPropertyEvidence.scala [hunk 3]
```diff
@@ -46,7 +46,12 @@ class UploadPropertyEvidence @Inject()(override val authenticated: Authenticated
   extends FileUploadController(authenticated, withLinkingSession, businessRatesAttachmentService) {
 
   def show() = withLinkingSession { implicit request =>
-    Ok(uploadEvidence(request.ses.submissionId, List.empty, request.ses.uploadEvidenceData.attachments.getOrElse(Map.empty), form))}
+    Ok(uploadEvidence(
+      request.ses.submissionId, List.empty,
```
Comment (DanielConnelly) [inline app/controllers/propertyLinking/UploadPropertyEvidence.scala]: can we name these parameters.

File: app/views/helpers/inputSelect.scala.html [hunk 4]
```diff
@@ -43,7 +43,7 @@
             <option>@{args.toMap.get('_emptyValueText)}</option>
         }
         @options.map { v =>
-            <option id="@(id)-@{v.replaceAll(" ", "-")}" value="@v" @if(value.exists(_.toLowerCase == v)){ selected } >@Messages(s"$id.${v.replaceAll(" ", "")}")</option>
+            <option id="@(id)-@{v.replaceAll(" ", "-")}" value="@v" @if(value.exists(_.equalsIgnoreCase(v))){ selected } >@Messages(s"$id.${v.replaceAll(" ", "")}")</option>
```
Comment (DanielConnelly) [inline app/views/helpers/inputSelect.scala.html]: is there a hMRC inputselect component we can use that already does this?

Comment (reddy-yattapu) [reply app/views/helpers/inputSelect.scala.html]: I am looking at uk.gov.hmrc.play.views.html.helpers._  I see input and inputRadioGroup but not inputSelect

**Comment by DanielConnelly** [comment]
LGTM 👍

**Comment by habeeb01** [comment]
👍

**Pull request merged**